### DISCUSSION TITLE
关于云角色卡的讨论

### DISCUSSION THREAD

**Discussion by MintCider** [sealdice/sealdice-core#840] [Ideas]
@fy0 @JustAnotherID @Xiangze-Li @MX-fox @MintCider 

1. 关于云角色卡的同步机制
2. 关于云角色卡的指令设计

**Comment by MintCider** [comment]
## 关于云角色卡的指令设计

新指令：

```text
.pc online
.pc offline
.pc cloudlist
```

### `.pc online`

* 当前使用群内独立卡（即未绑卡时）时报错：
  > 只有群聊无关的角色卡可以进行同步
* 当前绑定卡无 UUID 时报错：
  > 需要同步的角色卡必须拥有xx（见下文）
* 当前 UUID 云端不存在时：推送本角色卡数据至云端，并开启同步
* 当前 UUID 云端存在时：无冲突时开启同步，有冲突时处理冲突（取决于同步设计）

### `.pc offline`

* 当前使用群内独立卡（即未绑卡时）时报错：
  > 只有群聊无关的角色卡可以进行同步
* 当前绑定卡无 UUID 时报错：
  > 需要同步的角色卡必须拥有xx（见下文）
* 当前 UUID 云端不存在时：无事可做
* 当前 UUID 云端存在时：检查是否正在同步，然后停止同步或报错

### `.pc cloudlist`

展示当前用户所有云端角色卡

## 关于角色卡的 UUID

使用什么作为每个用户角色卡的 UUID 呢？角色卡名字看起来是个比较合理的方案。但是这也意味着相同用户在云端不能有两张同名角色卡。

**Comment by Xiangze-Li** [comment]
Ref sealdice/sealdice-core#278
